### PR TITLE
Feat/arbitrage component

### PR DESCRIPTION
**Pull request by cardosofede**
**Before submitting this PR, please make sure**:

- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
- Create Arbitrage component that works between DEX/CEX
- Create script that uses the arbitrage component

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

File: hummingbot/smart_components/arbitrage_executor/arbitrage_executor.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+                self.logger().error("Error calculating profitability", e)
+        elif self.arbitrage_status == ArbitrageExecutorStatus.ACTIVE_ARBITRAGE:
+            if self._cumulative_failures > self.max_retries:
+                self.arbitrage_status = ArbitrageExecutorStatus.FAILED
+                self.terminate_control_loop()
+            else:
+                self.check_order_status()
+
+    def check_order_status(self):
+        if self.buy_order.order.is_filled and self.sell_order.order.is_filled:
```
Comment (fengtality) [inline hummingbot/smart_components/arbitrage_executor/arbitrage_executor.py:123]:
```
2023-07-09 10:57:50,820 - 25875 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: 'NoneType' object has no attribute 'is_filled'
Traceback (most recent call last):
  File "/Users/feng/Code/hummingbot/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/Users/feng/Code/hummingbot/hummingbot/smart_components/smart_component_base.py", line 72, in control_loop
    await self.control_task()
  File "/Users/feng/Code/hummingbot/hummingbot/smart_components/arbitrage_executor/arbitrage_executor.py", line 120, in control_task
    self.check_order_status()
  File "/Users/feng/Code/hummingbot/hummingbot/smart_components/arbitrage_executor/arbitrage_executor.py", line 123, in check_order_status
    if self.buy_order.order.is_filled and self.sell_order.order.is_filled:

```
the above error is raised when there's no order object in this line

**Comment by cardosofede** [comment]
Yeah, I fixed it by adding in the conditional the check that the order
exists.

**Comment by rapcmia** [comment]
PR update:
- Test latest commit `6c76c7f0e`
- Overview on PR and script `arbitrage_with_smart_component`
- Setup this PR and latest gateway development branch
- Modified script using georli and paper trade.
- Start and observed behavior:
   - At the start of the script `getting error on price..`
     image
   - Getting `unexpected error while processing even 200`  when using to `paper_trade` and not related to current design of the PR ✅ 
   - Tested with Binance mainnet instead 👀 

---

After successfully closing the arbitrage, UI seems to have been broken
image
  - Run `status` command will return the same UI behavior
  - Compare trade data from client and goerli etherscan,
     - Txn recorded ok and no error found on the process 


Pending:
- Not able to fully test the script with status --live UI seems to be broken atm (3/3 attempts)

**Comment by fengtality** [comment]
Closed in favor of #6505

**Pull request closed without merging**